### ISSUE TITLE
mAP decreasing as iterations increase

### ISSUE THREAD

**Issue by HamzahNizami** [AlexeyAB/darknet#2744]
Hi, I've been training yolo-tinyv3 on my custom dataset. I ran for 4000 iterations and it resulted in the average loss lowest being around 2.1 and the mAP at default threshold to be at 74.66%. I let the model continue to run to 25k iterations. At this point, my average loss has decreased to 0.9 and my mAP has also decreased to 56.1%. I was hoping someone could explain to me why this may be happening. 

Thanks in advance

**Comment by AlexeyAB** [comment]
@HamzahNizami Hi,

Do you check mAP on the Training or Validation dataset?
How many images are in your Training and Validation datasets?
Can you rename your cfg-file to txt-file and attach it to your message?

**Comment by HamzahNizami** [comment]
I believe it is testing it on the validation set. How do I ensure this is the case? 

My training set has 754 images and my validation set has 88.

**Comment by AlexeyAB** [comment]
> I believe it is testing it on the validation set. How do I ensure this is the case?

Check that train=train.txt and valid=valid.txt in your `obj.data` file.

Don't set subdivisions more than batch in cfg. Since `mini_batch=batch/subdivisions`

It seems your met overfitting, since you have low number of training images: https://github.com/AlexeyAB/darknet#when-should-i-stop-training

**Comment by HamzahNizami** [comment]
Yes, it is set to valid.txt. 

Okay, what so should I lower subdivisions or increase batch? 

Okay, I was thinking this may be the case.

Thanks for the rapid response its very helpful!

**Comment by LukeAI** [comment]
I appear to be getting a decreasing mAP over time quite early. Is my model overfitting already? @AlexeyAB 

My validation set is 1544 images (oversampled to 2931) and my training set is 59815 images (oversampled to 233554 images).
I am using mixed precision on a 2080Ti with SGDR enabled.

![chart](https://user-images.githubusercontent.com/43993778/57014307-70875900-6c07-11e9-84cc-236a928a9665.png)

```
batch=64
subdivisions=32
width=608
height=608
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=sgdr
steps=400000,450000
scales=.1,.1
```
